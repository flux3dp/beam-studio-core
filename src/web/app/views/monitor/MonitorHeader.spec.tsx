/* eslint-disable import/first */
import * as React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

jest.mock('app/contexts/MonitorContext', () => ({
  MonitorContext: React.createContext(null),
}));

import MonitorHeader, { NavBtnType } from './MonitorHeader';

describe('should render correctly', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  test('NavBtnType is BACK', () => {
    const onClose = jest.fn();
    const onNavigationBtnClick = jest.fn();
    MonitorHeader.contextType = React.createContext({
      onClose,
      onNavigationBtnClick,
    });
    const wrapper = mount(<MonitorHeader
      name="myDevice"
      navBtnType={NavBtnType.BACK}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper.find('div.close').simulate('click');
    expect(onClose).toHaveBeenCalledTimes(1);

    wrapper.find('div.back').simulate('click');
    expect(onNavigationBtnClick).toHaveBeenCalledTimes(1);
  });

  test('NavBtnType is FOLDER', () => {
    const onNavigationBtnClick = jest.fn();
    MonitorHeader.contextType = React.createContext({
      onClose: jest.fn(),
      onNavigationBtnClick,
    });
    const wrapper = mount(<MonitorHeader
      name="myDevice"
      navBtnType={NavBtnType.FOLDER}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper.find('div.back').simulate('click');
    expect(onNavigationBtnClick).toHaveBeenCalledTimes(1);
  });

  test('NavBtnType is something else', () => {
    MonitorHeader.contextType = React.createContext({
      onClose: jest.fn(),
      onNavigationBtnClick: jest.fn(),
    });
    const wrapper = mount(<MonitorHeader
      name="myDevice"
      navBtnType={NavBtnType.NONE}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
