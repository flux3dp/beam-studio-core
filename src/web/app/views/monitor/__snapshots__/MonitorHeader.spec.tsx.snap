// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly NavBtnType is BACK 1`] = `
<MonitorHeader
  name="myDevice"
  navBtnType={1}
>
  <div
    className="header"
  >
    <div
      className="title"
    >
      <span>
        myDevice
      </span>
      <div
        className="close"
        onClick={[MockFunction]}
      >
        <div
          className="x"
        />
      </div>
      <div
        className="back"
        onClick={[MockFunction]}
      >
        <i
          className="fa fa-angle-left"
        />
      </div>
    </div>
  </div>
</MonitorHeader>
`;

exports[`should render correctly NavBtnType is FOLDER 1`] = `
<MonitorHeader
  name="myDevice"
  navBtnType={2}
>
  <div
    className="header"
  >
    <div
      className="title"
    >
      <span>
        myDevice
      </span>
      <div
        className="close"
        onClick={[MockFunction]}
      >
        <div
          className="x"
        />
      </div>
      <div
        className="back"
        onClick={[MockFunction]}
      >
        <img
          src="img/folder.svg"
        />
      </div>
    </div>
  </div>
</MonitorHeader>
`;

exports[`should render correctly NavBtnType is something else 1`] = `
<MonitorHeader
  name="myDevice"
  navBtnType={0}
>
  <div
    className="header"
  >
    <div
      className="title"
    >
      <span>
        myDevice
      </span>
      <div
        className="close"
        onClick={[MockFunction]}
      >
        <div
          className="x"
        />
      </div>
      <div />
    </div>
  </div>
</MonitorHeader>
`;
